src/Umbraco.Web/Routing/ContentFinderByRedirectUrl.cs

```diff
@@ -52,8 +52,12 @@ public bool TryFindContent(PublishedContentRequest contentRequest)
                 return false;
             }
 
+            // Apending any querystring from the incoming request to the redirect url.
+            url = string.IsNullOrEmpty(contentRequest.Uri.Query) ? url : url + contentRequest.Uri.Query;
```

Comment:
A more readable version would be:
```c#
var query = contentRequest.Uri.Query;
if (!string.IsNullOrEmpty(query))
{
    url = url + query;
}
```

Reply:
Hmm, might be more readable depending on who you ask. For the code in the PR is just as readable but if someone wants to change it I'm all fine with that =D